### PR TITLE
Fix mach0 endianness

### PR DESCRIPTION
**Pull request by wargio**
# DO NOT SQUASH

**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Tries to fix mach0 endianness

### CODE REVIEW

**Comment by thestr4ng3r** [comment]
For endian `rz_buf_fread_at` is usually not the issue since it is aware of the endian. But I still like seeing it removed since it is very easily misused and evades type-checking.

**Comment by wargio** [comment]
yeah, it doesn't fix the endianness :/
i think the issue is on maps, but i don't know where.

**Comment by thestr4ng3r** [comment]
No worries, I think I already found it, will check in a moment.

**Comment by thestr4ng3r** [comment]
That seems to fix half of them, but please don't merge yet since I want to test it on a ppc macho.

**Comment by thestr4ng3r** [comment]
This does it at least on sparc64. I have seen two more failures in `db/formats/mach0/fatmach0` on ppc Mac OS X, but that can be addressed separately. Not sure if they happen on system-z too.

**Review by thestr4ng3r — APPROVED** [review]

File: librz/bin/format/mach0/mach0_relocs.c [hunk 1]
```diff
@@ -10,6 +10,37 @@
...
 
+#define RELOCATION_INFO_SIZE 8 // sizeof(struct relocation_info)
+
+/**
+ * \param buf buffer of at least RELOCATION_INFO_SIZE bytes
+ */
+static void read_relocation_info(struct relocation_info *dst, ut8 *src, bool big_endian) {
+	dst->r_address = (st32)rz_read_at_ble32(src, 0, big_endian);
+	uint32_t field = rz_read_at_ble32(src, 4, big_endian);
+	if (big_endian) {
```
Comment (XVilka) [inline librz/bin/format/mach0/mach0_relocs.c:21]: QEMU has helper macroses like `EXTRACT` etc. We could think about something similar too.

**Review by XVilka — APPROVED** [review]
Looks good. Lets see what Travis says.

**Comment by XVilka** [comment]
17 XX now.

**Pull request merged**